sycl/include/sycl/ext/intel/experimental/kernel_properties.hpp

```diff
@@ -10,72 +10,58 @@
 
 #pragma once
 
-#include <sycl/ext/intel/experimental/detail/misc_intrin.hpp>
-
-#include <sycl/detail/boost/mp11.hpp>
+#include <sycl/ext/oneapi/properties/property.hpp>
+#include <sycl/ext/oneapi/properties/property_value.hpp>
 
 namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
-namespace ext::intel::experimental {
-
-namespace kernel_properties {
-
-/// <summary>
-///  This namespace contains APIs to set kernel properties.
-/// </summary>
+namespace ext {
+namespace intel {
+namespace experimental {
 
-// Implementation note: <property_class>::value fields should match property IDs
-// specified in llvm/lib/SYCLLowerIR/LowerKernelProps.cpp
-
-namespace detail {
-// Proxy to access private property classes' fields from the API code.
-template <class T> struct proxy {
-  static constexpr int value = T::value;
+enum class RegisterAllocMode : uint32_t {
+  AUTO = 0,
+  LARGE = 2,
 };
-} // namespace detail
 
-/// A boolean property which requests the compiler to use large register
-/// allocation mode at the expense of reducing the amount of available hardware
-/// threads.
-struct use_large_grf_tag {
-  template <class T> friend struct detail::proxy;
-
-private:
-  // Property identifier
-  static constexpr int value = 0;
+struct register_alloc_mode_key {
```

Comment:
Is this API a part of some spec? If not, we should not expose this publicly (and should hide into detail:: namespace).

Reply:
@steffenlarsen @dm-vodopyanov let's have the discussion about the spec/namespace here if you don't mind.

There's an internal tracker on the SYCL language group to make a spec for this, I will send it to you both , but we don't have it today.

Right now this is in the `intel::experimental` namespace. I'm happy to move it to one that is more clear that there's more spec, but I thought `experimental` was that namespace. Can you give me a suggestion? Should it just be `intel::experimental::detail`? Thanks

Reply:
If it is just internal, we could just have it in `sycl::detail`. However, if we are planning on adding an extension of this, it seems a little weird to have an internal implementation first. Do we plan on using it internally until we have an extension for it?

Reply:
Yes that's right. We need to use this internally because of the large performance impact, but the spec is supposed to be relevant to all (non-intel) hardware, so we don't have the extension spec yet.

Reply:
In that case, I think having them in `sycl::detail` for now should be alright, then we can promote them out and adjust when we decide on the extension interface.

Reply:
cool thx, let me implement that